lib/app/utils/validation.rb

```diff
@@ -192,5 +192,21 @@ def self.guess_r4_profile(resource)
     def self.observation_contains_code(observation_resource, code)
       observation_resource&.code&.coding&.any? { |coding| coding&.code == code }
     end
+
+    def self.resource_contains_category(resource, category_code)
```

Comment:
Would it make more sense to have the signature be `def self.resource_contains_category(resource, category_system, category_code)` so you don't need to have that `if` statement that will need to be updated if there is ever another resource type?

Reply:
I had that version before this one. Then I think why I force the caller to provide category_system while the function itself can decide. It is a problem when the binding value set has codes from multiple code system. At least for Observation and DiagnosticReport, both binding ValueSets have single code system value.

Reply:
I fixed it.